session-12/task/ctx_pkg/task1.go

```diff
@@ -0,0 +1,31 @@
+package ctx_pkg
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+func longRunningProgram() {
+	for i := 1; i <= 10; i++ {
+		fmt.Println(i)
+		time.Sleep(time.Second)
+	}
+}
+
+func cancelFuncTimeout() {
+	ctx, cancel := context.WithTimeout(context.Background(), (3*time.Second)-(time.Millisecond)) // To achieve consistent results across executions.
+	defer cancel()
+	go longRunningProgram()
+	for {
+		select {
+		case <-ctx.Done():
+			fmt.Println("(cancellation)")
+			return
+		}
+	}
```

Comment:
TimeOut duzgun helldir, sadece taskin predmetine gore imitasiya etmek isteyirikse sırf withCancel o zaman
```suggestion
func cancelFuncTimeout() {
	_, cancel := context.WithCancel(context.Background())
	defer cancel()
	go longRunningProgram()
	for {
		select {
		case <-time.After(3 * time.Second):
			fmt.Println("(cancellation)")
			return
		}
	}
```